packages/data/src/registry.js

```diff
@@ -88,6 +90,61 @@ export function createRegistry( storeConfigs = {}, parent = null ) {
 		return parent && parent.select( reducerKey );
 	}
 
+	const getResolveSelectors = memize(
+		( selectors ) => {
+			return mapValues(
+				omit(
+					selectors,
+					[
+						'getIsResolving',
+						'hasStartedResolution',
+						'hasFinishedResolution',
+						'isResolving',
+						'getCachedResolvers',
+					]
+				),
+				( selector, selectorName ) => {
+					return ( ...args ) => {
+						return new Promise( ( resolve ) => {
+							const hasFinished = () => selectors
+								.hasFinishedResolution( selectorName, args );
+							const getResult = () => selector.apply( null, args );
+
+							// trigger the selector (to trigger the resolver)
+							const result = getResult();
+							if ( hasFinished() ) {
+								return resolve( result );
+							}
+
+							const unsubscribe = subscribe( () => {
+								if ( hasFinished() ) {
+									unsubscribe();
+									resolve( getResult() );
+								}
+							} );
+						} );
+					};
+				}
+			);
+		},
+		{ maxSize: 1 }
+	);
+
+	/**
+ 	 * Given the name of a registered store, returns an object containing the store's
```

Comment:
There is an ESLint warning here caused by mixed whitespace at the beginning of the line.

```
/Users/andrew/Documents/Code/gutenberg/packages/data/src/registry.js
  134:0  warning  Expected JSDoc block to be aligned  jsdoc/check-alignment
```

See also: #18458